neuroanalysis/event_detection.py

```diff
@@ -95,45 +95,175 @@ def zero_crossing_events(data, min_length=3, min_peak=0.0, min_sum=0.0, noise_th
-    Optionally adjusts index to an extrapolated baseline-crossing.
+    Finds regions in a trace that cross a threshold value (as measured by distance from baseline) and then 
+    recross threshold (bumps).  If a threshold is crossed at the end of the trace, an event may be excluded
+    or the beginning/end may be used as the the start/end of the event (depending on the value of *omit_ends*). 
+    Optionally adjusts start and end index of an event to an extrapolated baseline-crossing and calculates 
+    values. 
+
+    Parameters                                                                                                                                     
+    ==========
+    trace: *Trace* instance                            
+    threshold: float or np.array with dimensions of *trace.data*
+        algorithm checks if waveform crosses both positive and negative thresholds.  
+        i.e. if -5. is provided, the algorithm looks for places where the waveform crosses +/-5.
+        If an array is provided the *threshold* is dynamic
+
+    adjust_times: boolean
+        if True, move the start and end times of the event outward, estimating the zero-crossing point for the event
+    
+    Returns
+    =======
+    events: numpy structured array.  
+        An event is a region of the *Trace.data* waveform that crosses above *threshold* and then falls below threshold again
+        Sometimes referred to as a 'bump'.  There are additional criteria (not listed here) for a bump to be considered an event.    
+        Each index contains information about an event.  Fields as follows:
+        index: int
+           index of the initial crossing of the *threshold*
+        len: int
+            index length of the event
+        sum: float
+           sum of the values in the array between the start and end of the event
+        peak: float
+           peak value of event
+        peak_index: int
+           index value of the peak
+        time: float, or np.nan if timing not available
+           time of the onset of an event
+        duration: float, or np.nan if timing not available
+           duration of time of the event
+        area: float, or np.nan if timing not available
+           area under the curve of the event
+        peak_time: float, or np.nan if timing not available
+           time of peak
     """
-    threshold = abs(threshold)
+
+
     data = trace.data
     data1 = data - baseline
+    # convert threshold array
+    if isinstance(threshold, float):
+        threshold = np.ones(len(data)) * abs(threshold)
+
+
     #if (hasattr(data, 'implements') and data.implements('MetaArray')):
     
-    ## find all threshold crossings
-    masks = [(data1 > threshold).astype(np.byte), (data1 < -threshold).astype(np.byte)]
+    ## find all positive and negative threshold crossings of baseline adjusted data
+
+    if debug:
```

Comment:
I'm not so keen on merging debugging code unless there's a good reason to keep it here?

Reply:
Yes I figured, you would want it removed, but I didn't want to do it before you looked at the rest of the code incase I needed to look at stuff again.  The debug plots are particularly insightful for your questions below about removing the +1 on line 191.  Once you are happy it can be removed before merge along with the 'debug' argument into the functions.